src/wallet/test/init_tests.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <boost/test/unit_test.hpp>
+
+#include <test/test_bitcoin.h>
+#include <wallet/test/init_test_fixture.h>
+
+#include <wallet/wallet.h>
+#include <wallet/init.cpp>
```

Comment:
Why this include cpp file?

Reply:
The `WalletInit` class was moved from the header file to the cpp file in this pull request https://github.com/bitcoin/bitcoin/pull/12836

Should I move it back or is there a bigger picture architectural issue that I'm not seeing?

Reply:
Include `src/walletinitinterface.h`?

Reply:
🤔 how would I create an instance of the `WalletInit` class to test its implementation of `Verify`?

(I'll take "read this book / article" as an answer if I'm out of my depth here haha)

Reply:
The program will find the implementation by C++ virtual method table?

Reply:
There is already an instance. Just replace the include.

Reply:
Ah ok, fixed, I added `init.h` as well as it has the global instance in it